src/Workspaces/Core/Portable/Workspace/Solution/ProjectState_Checksum.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Serialization;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal partial class ProjectState
+    {
+        public bool TryGetStateChecksums(out ProjectStateChecksums stateChecksums)
+        {
+            return _lazyChecksums.TryGetValue(out stateChecksums);
+        }
+
+        public async Task<Checksum> GetChecksumAsync(CancellationToken cancellationToken)
```

Comment:
No.  i mean it literally doesn't need to be 'async'.  It can return a ```Task<Checksum>```.  But you don't need to use async/await.